client/widgets/Buttons.cpp

```diff
@@ -447,26 +447,79 @@ void CToggleGroup::selectionChanged(int to)
 		parent->redraw();
 }
 
-void CToggleGroup::show(SDL_Surface * to)
+CVolumeSlider::CVolumeSlider(const Point &position, const std::string &defName, const int value,
+                             const std::pair<std::string, std::string> * const help) :
+	value(value),
+	animImage(new CAnimImage(new CAnimation(defName), 0, 0, position.x, position.y)),
+	helpHandlers(help)
 {
-	if (musicLike)
+	OBJ_CONSTRUCTION_CAPTURING_ALL;
+	assert(!defName.empty());
+	addUsedEvents(LCLICK | RCLICK | WHEEL);
+	pos.x += position.x;
+	pos.y += position.y;
+	pos.w = (animImage->pos.w + 1) * animImage->size();
+	pos.h = animImage->pos.h;
+	type |= REDRAW_PARENT;
```

Comment:
I had no choice but to add this flag because the area wouldn't refresh in redraw.
If there is a way to redraw only the contents of pos, I would like to know.